dotnet/src/Connectors/Connectors.OpenAI/OpenAIPromptExecutionSettings.cs

```diff
@@ -254,6 +254,39 @@ public int ResultsPerPrompt
         }
     }
 
+    /// <summary>
+    /// Whether to return log probabilities of the output tokens or not.
+    /// If true, returns the log probabilities of each output token returned in the `content` of `message`.
+    /// </summary>
+    [Experimental("SKEXP0010")]
+    [JsonPropertyName("logprobs")]
+    public bool? Logprobs
+    {
+        get => this._logprobs;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._logprobs = value;
+        }
+    }
+
+    /// <summary>
+    /// An integer specifying the number of most likely tokens to return at each token position, each with an associated log probability.
```

Comment:
Any reason not t use the description from the OpenAI documentation?

`An integer between 0 and 20 specifying the number of most likely tokens to return at each token position, each with an associated log probability. logprobs must be set to true if this parameter is used.`

Reply:
Mostly because of integer range difference. OpenAI documentation says `integer between 0 and 20`, while Azure .NET SDK says `integer between 0 and 5`:
https://github.com/Azure/azure-sdk-for-net/blob/576ace699a0b61721762d5898e6d6a201d4861a0/sdk/openai/Azure.AI.OpenAI/src/Generated/ChatCompletionsOptions.cs#L195
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/bf58b8ea-0892-4d29-9d63-b0ceccea88c9)


And I tested both limitations using OpenAI and Azure OpenAI endpoints and they work as mentioned in documentation. Taking into account that we use `OpenAIPromptExecutionSettings` for both OpenAI and Azure OpenAI, I decided to not mention that limitation. But we can specify both limitations if needed. We can also add `logprobs must be set to true if this parameter is used.` part.

Let me know what you think about it.

Reply:
Thanks, that's good to know